### PR TITLE
CEDS-1681 Audit Movement Submissions

### PR DESCRIPTION
**Pull request by PauloPortugal**
CEDS-1681 Audit Movements DUCR Association Submission
CEDS-1683 Audit Movements DUCR Disassociate Submission
CEDS-1684 Audit Shut a DUCR Submission

### CODE REVIEW

File: app/services/SubmissionService.scala [hunk 1]
```diff
@@ -73,51 +76,60 @@ class SubmissionService @Inject()(
...
     connector
       .sendDisassociationRequest(buildDisassociationRequest(disassociateDucr.ducr).toString)
       .map(_.status)
       .andThen {
-        case Success(_) =>
+        case Success(status) =>
+          auditService.audit(
+            Choice(Choice.AllowedChoiceValues.DisassociateDUCR),
+            auditService
+              .auditDisassociateData(request.user.eori, disassociateDucr.ducr, status.toString)
```
Comment (ed0906) [inline app/services/SubmissionService.scala]: Would it not make sense to have `auditService.auditDisassociateData(...)` actually call `auditService.audit` under the hood?

File: app/services/SubmissionService.scala [hunk 2]
```diff
@@ -73,51 +76,60 @@ class SubmissionService @Inject()(
...
+    shutMucr: ShutMucr
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext, request: AuthenticatedRequest[_]): Future[Int] = {
     val timer = metrics.startTimer(Choice.AllowedChoiceValues.ShutMucr)
     connector.sendShutMucrRequest(buildShutMucrRequest(shutMucr.mucr).toString).map(_.status).andThen {
-      case Success(_) =>
+      case Success(status) =>
+        auditService.audit(
+          Choice(Choice.AllowedChoiceValues.ShutMucr),
+          auditService
+            .auditShutMucrData(request.user.eori, shutMucr.mucr, status.toString)
```
Comment (ed0906) [inline app/services/SubmissionService.scala]: Similar here

File: app/services/audit/AuditService.scala [hunk 3]
```diff
@@ -26,15 +26,39 @@ import uk.gov.hmrc.play.audit.AuditExtensions
...
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class AuditService @Inject()(connector: AuditConnector, @Named("appName") appName: String)(
   implicit ec: ExecutionContext
 ) {
-
   private val logger = Logger(this.getClass)
 
+  def auditShutMucrData(eori: String, mucr: String, result: String): Map[String, String] =
```
Comment (SteveSugden) [inline app/services/audit/AuditService.scala]: Seems a bit odd calling these methods to create the data then having to call "audit" with the data and choice.  Why not have these "data builder" methods call "audit" with the relevant choice

Comment (SteveSugden) [reply app/services/audit/AuditService.scala]: Then maybe "audit" can be private (if it's not called/used elsewhere)?

File: app/services/SubmissionService.scala (hunk 1 above)
Comment (PauloPortugal) [reply app/services/SubmissionService.scala]: Very good point! @PatrykRudnicki suggested just that as well

File: app/services/audit/AuditService.scala (hunk 3 above)
Comment (PauloPortugal) [reply app/services/audit/AuditService.scala]: That is a very good point. It seems that everyone got that same conclusion.
The `auditShutMucrData` `auditDisassociateData` were in my first iteration part of the controller/service layers. I then refactored them to put all inside the `AuditService`. :)
Excellent advice.

Comment (PauloPortugal) [reply app/services/audit/AuditService.scala]: This has been addressed.

File: app/services/SubmissionService.scala (hunk 1 above)
Comment (PauloPortugal) [reply app/services/SubmissionService.scala]: This has been addressed

File: app/services/SubmissionService.scala (hunk 2 above)
Comment (PauloPortugal) [reply app/services/SubmissionService.scala]: This has been addressed.

File: app/services/SubmissionService.scala [hunk 4]
```diff
@@ -73,51 +73,49 @@ class SubmissionService @Inject()(
...
+          )
           timer.stop()
           metrics.incrementCounter(Choice.AllowedChoiceValues.AssociateDUCR)
       }
   }
 
   def submitDucrDisassociation(
     disassociateDucr: DisassociateDucr
-  )(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Int] = {
+  )(implicit hc: HeaderCarrier, ec: ExecutionContext, request: AuthenticatedRequest[_]): Future[Int] = {
```
Comment (PatrykRudnicki) [inline app/services/SubmissionService.scala]: I'm not sure that you need to have request here

**Pull request merged**